backend/de.metas.monitoring/src/main/java/de/metas/monitoring/adapter/MicrometerPerformanceMonitoringService.java

```diff
@@ -42,63 +42,88 @@
 public class MicrometerPerformanceMonitoringService implements PerformanceMonitoringService
 {
 	private final MeterRegistry meterRegistry;
-	private final Optional<APMPerformanceMonitoringService> apmPerformanceMonitoringService;
+	private static final ThreadLocal<PerformanceMonitoringData> perfMonDataTL = ThreadLocal.withInitial(PerformanceMonitoringData::new);
```

Comment:
why u create a new instance as initial value?

are you aware that you will get a new instance everytime a thread is created in our system (and we have quite a lot) even if u need this performance monitoring or not?

suggestion: can't we go with null initial value and create the instance only when it's needed?